Sources/PackageGraph/Resolution/ResolvedPackage.swift

```diff
@@ -64,8 +64,41 @@ public struct ResolvedPackage {
         platformVersionProvider: PlatformVersionProvider
     ) {
         self.underlying = underlying
-        self.targets = targets
-        self.products = products
+
+        var processedTargets = IdentifiableSet<ResolvedTarget>(targets)
+        var processedProducts = [ResolvedProduct]()
+        // Make sure that direct macro dependencies of test products are also built for the target triple.
+        // Without this workaround, `assertMacroExpansion` in tests can't be built, as it requires macros
+        // and SwiftSyntax to be built for the target triple: https://github.com/apple/swift-package-manager/pull/7349
+        for var product in products {
+            if product.type == .test {
+                var targets = IdentifiableSet<ResolvedTarget>()
+                for var target in product.targets {
+                    var dependencies = [ResolvedTarget.Dependency]()
+                    for dependency in target.dependencies {
+                        switch dependency {
+                        case .target(var target, let conditions) where target.type == .macro:
+                            target.buildTriple = .destination
+                            dependencies.append(.target(target, conditions: conditions))
+                            processedTargets.insert(target)
```

Comment:
Just to clarify: If the destination is the same as the host, we end up with a duplicate dependency declaration, right? Is that or could that be an issue?

Reply:
It's not necessarily a complete duplicate, as in the future it allows us to build tools (plugins and macros) with different parameters than destination products. For example, we could have macros and SwiftSyntax used in macros built in release mode, but end products (and for example another instance of SwiftSyntax used in end products) built in debug mode. If build parameters are exactly the same, we could apply an optimization and build only once, but as agreed with Boris previously, this optimization can be implemented in a separate PR.

Reply:
OK, just wanted to make sure this wouldn’t cause any issues.